Sources/OpenAPIURLSession/URLSessionBidirectionalStreaming/HTTPBodyOutputStreamBridge.swift

```diff
@@ -0,0 +1,286 @@
+        dispatchPrecondition(condition: .onQueue(Self.streamQueue))
+        switch action {
+        case .none: return
+        case .resumeProducer(let producerContinuation):
+            producerContinuation.resume()
+            performAction(self.state.resumedProducer())
+        case .writeBytes(let chunk): writePendingBytes(chunk)
+        case .cancelProducerAndCloseStream(let producerContinuation):
+            producerContinuation.resume(throwing: CancellationError())
+            outputStream.close()
+        case .cancelProducer(let producerContinuation): producerContinuation.resume(throwing: CancellationError())
+        case .closeStream: outputStream.close()
+        }
+    }
+
+    func startWriterTask() {
+        dispatchPrecondition(condition: .onQueue(Self.streamQueue))
+        let task = Task {
+            dispatchPrecondition(condition: .notOnQueue(Self.streamQueue))
+            for try await chunk in httpBody {
+                try await withCheckedThrowingContinuation { continuation in
+                    Self.streamQueue.async {
+                        debug("Output stream delegate produced chunk and suspended producer.")
+                        self.performAction(self.state.producedChunkAndSuspendedProducer(chunk, continuation))
+                    }
+                }
+            }
+            Self.streamQueue.async {
+                debug("Output stream delegate wrote final chunk.")
+                self.performAction(self.state.wroteFinalChunk())
+            }
+        }
+        self.performAction(self.state.startedProducerTask(task))
+    }
+
+    private func writePendingBytes(_ bytesToWrite: Chunk) {
+        dispatchPrecondition(condition: .onQueue(Self.streamQueue))
+        precondition(!bytesToWrite.isEmpty)
+        guard outputStream.streamStatus == .open else {
+            debug("Output stream closed unexpectedly.")
+            performAction(self.state.wroteBytes(numBytesWritten: 0, streamStillHasSpaceAvailable: false))
+            return
+        }
+        switch bytesToWrite.withUnsafeBytes({ outputStream.write($0.baseAddress!, maxLength: bytesToWrite.count) }) {
+        case 0:
+            debug("Output stream delegate reached end of stream when writing.")
+            performAction(self.state.endEncountered())
+        case -1:
+            debug("Output stream delegate encountered error writing to stream: \(outputStream.streamError!).")
+            performAction(self.state.errorOccurred(outputStream.streamError!))
+        case let written where written > 0:
+            debug("Output stream delegate wrote \(written) bytes to stream.")
+            performAction(
+                self.state.wroteBytes(
+                    numBytesWritten: written,
+                    streamStillHasSpaceAvailable: outputStream.hasSpaceAvailable
+                )
+            )
+        default: preconditionFailure()
```

Comment:
Can you add a string to the crash (here and in the rest of the file)?

Reply:
Added message to all the `precondition` and `preconditionFailure` calls throughout (except a few in tests).